File: Monte-carlo.c
#include<stdio.h>
#include<stdlib.h>
#include<time.h>
double function(double x){
	// function which has be integrated
	return x*x;
}

int main(){
	long iterations;
	double average,upperlimit,lowerlimit,domain,randd;
	//seeding random number with time
	srand(time(NULL));
	
	
	//Input the number of iterations, more the iterations are, better the result will be
	//10^6 or more would be good.

	printf("Iterations : ");
	scanf("%ld",&iterations);	
	printf("Lower limit : ");
	scanf("%lf",&lowerlimit);
	printf("Upper limit : ");
	scanf("%lf",&upperlimit);
	
	
	domain = upperlimit-lowerlimit;
	average = 0;
	for(int i=0;i<iterations;i++){
		randd = lowerlimit + (double)(rand())/RAND_MAX*domain;
		average += function(randd);	
	}
	printf("Value : %.10f\n",average*domain/iterations);

}

File: Monte-carlo.py
from numpy import random as r

# Function which has to be integrated
function = lambda x : x**2

#More the better, but ~10^6 or more would work 
iteration   = int(input("Iterations : "))
#Input lower limit
lowerlimit = float(input("Lower limit : "))
#Input upper limit
upperlimit = float(input("Upper limit : "))
domain = upperlimit-lowerlimit


average = 0
for i in range(iteration):
	random = lowerlimit + r.random()*domain
	'''
	random = r.uniform(lowerlimit,upperlimit)	#would also work
	'''	
	average+=function(random)
print("Value : ",average*domain/iteration)






